crates/query-engine/translation/src/translation/query/filtering.rs

```diff
@@ -67,116 +68,153 @@ pub fn translate_expression(
             joins.extend(left_joins);
-            let right = values
-                .iter()
-                .map(|value| {
-                    let (right, right_joins) = translate_comparison_value(
-                        env,
-                        state,
-                        root_and_current_tables,
-                        value.clone(),
-                        &typ,
-                    )?;
-                    joins.extend(right_joins);
-                    Ok(right)
-                })
-                .collect::<Result<Vec<sql::ast::Expression>, Error>>()?;
 
-            Ok((
-                sql::ast::Expression::BinaryArrayOperation {
-                    left: Box::new(left),
-                    operator: match operator {
-                        models::BinaryArrayComparisonOperator::In => {
-                            sql::ast::BinaryArrayOperator::In
-                        }
+            // eq is a built-in operator, so we define it here in case we need it.
+            let eq_op = query_engine_metadata::metadata::ComparisonOperator {
+                operator_name: "=".into(),
+                argument_type: left_typ.clone(),
+
+                is_infix: true,
+            };
+            let op = if operator == "eq" {
+                &eq_op
+            } else {
+                env.lookup_comparison_operator(&left_typ, operator)?
+            };
+            let (right, right_joins) = translate_comparison_value(
+                env,
+                state,
+                root_and_current_tables,
+                value.clone(),
+                database::Type::ScalarType(op.argument_type.clone()),
+            )?;
+            joins.extend(right_joins);
+
+            if op.is_infix {
+                Ok((
+                    sql::ast::Expression::BinaryOperation {
+                        left: Box::new(left),
+                        operator: sql::ast::BinaryOperator(op.operator_name.clone()),
+                        right: Box::new(right),
                     },
-                    right,
-                },
-                joins,
-            ))
+                    joins,
+                ))
+            } else {
+                Ok((
+                    sql::ast::Expression::FunctionCall {
+                        function: sql::ast::Function::Unknown(op.operator_name.clone()),
+                        args: vec![left, right],
+                    },
+                    joins,
+                ))
+            }
         }
 
         models::Expression::Exists {
             in_collection,
             predicate,
-        } => Ok((
-            translate_exists_in_collection(
-                env,
-                state,
-                root_and_current_tables,
-                in_collection.clone(),
-                predicate,
-            )?,
-            vec![],
-        )),
+        } => match predicate {
```

Comment:
predicates are now optional instead of being an equivalence of `true` when they are omitted.